### PR TITLE
PLINK2 add nf-tests and fix broken pytests

### LINKED ISSUE #3965: plink2/extract test is broken
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

The test of the `plink2/extract` test is broken since the addition of automatic compression (#1258). This is because the [test](https://github.com/nf-core/modules/blob/28091e156c2828a64b6d89227ac472f969d1c98b/tests/modules/nf-core/plink2/extract/main.nf#L16-L19) expects a plain text file from `plink2/vcf` to select a variant.

Note: also need to remove the `--make-pgen` option from `tests/modules/nf-core/plink2/extract/nextflow.config` as it's now already set in the module itself.

### Command used and terminal output

```console
$ nextflow run tests/modules/nf-core/plink2/extract -entry test_plink2_extract -c modules/tests/config/nextflow.config
N E X T F L O W  ~  version 23.04.1
Launching `../src/nf-core/modules/tests/modules/nf-core/plink2/extract/main.nf` [sharp_kalam] DSL2 - revision: 003f259eee
executor >  local (1)
[63/af4903] process > test_plink2_extract:PLINK2_VCF (test) [100%] 1 of 1 ✔
[-        ] process > test_plink2_extract:PLINK2_EXTRACT    -
executor >  local (1)
[63/af4903] process > test_plink2_extract:PLINK2_VCF (test) [100%] 1 of 1 ✔
[-        ] process > test_plink2_extract:PLINK2_EXTRACT    -
ERROR ~ Input length = 1

 -- Check '.nextflow.log' file for details
```

### Relevant files

[nextflow.log](https://github.com/nf-core/modules/files/12877825/nextflow.log)

### System information

- Nextflow 23.04.1 installed through Conda
- HPC
- Local executor
- Singularity
- Ubuntu 18.04

### PR DESCRIPTION
**Pull request by mauro-saporita**
## PR checklist

Closes #3965 

- [ ] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/plink2/vcf/main.nf [hunk 1]
```diff
@@ -13,7 +13,7 @@ process PLINK2_VCF {
     output:
     tuple val(meta), path("*.pgen")    , emit: pgen
     tuple val(meta), path("*.psam")    , emit: psam
-    tuple val(meta), path("*.pvar.zst"), emit: pvar
+    tuple val(meta), path("*.pvar"), emit: pvar
```
Comment (FriederikeHanssen) [inline modules/nf-core/plink2/vcf/main.nf:16]:
```suggestion
    tuple val(meta), path("*.pvar")    , emit: pvar
```

**Review by FriederikeHanssen — COMMENTED** [review]
Should both options be supported as optional outputs depending on the set args?

**Comment by priyanka-surana** [comment]
Add another value to `meta.yml` -- you have 2. From what I understand, the CI tests need minimum 3 to pass.

File: modules/nf-core/plink2/vcf/main.nf (hunk 1 above)
Comment (mauro-saporita) [reply modules/nf-core/plink2/vcf/main.nf]: Good point, just added

**Comment by priyanka-surana** [comment]
Might be best to wait till they finish restructuring. Ask one of the nf-test people if you need to bring that in. Based on the conversations, seems like they are turning off pytest.

**Review by sateeshperi — APPROVED** [review]

File: modules/nf-core/plink2/score/main.nf [hunk 2]
```diff
@@ -26,9 +26,8 @@ process PLINK2_SCORE {
     plink2 \\
         --threads $task.cpus \\
         --memory $mem_mb \\
-        --pfile ${pgen.baseName} vzs \\
-        --score ${scorefile} \\
-        $args \\
+        --pfile ${pgen.baseName} \\
+        --score ${scorefile} $args \\
```
Comment (Emiller88) [inline modules/nf-core/plink2/score/main.nf:30]:
```suggestion
        --score ${scorefile} \\
        $args \\
```

File: modules/nf-core/plink2/vcf/meta.yml [hunk 3]
```diff
@@ -3,6 +3,7 @@ description: Import variant genetic data using plink2
 keywords:
   - plink2
   - import
+  - variant genetic
```
Comment (Emiller88) [inline modules/nf-core/plink2/vcf/meta.yml:6]:
```suggestion
  - "variant genetic"
```

Not sure this won't break things @mashehu ?

**Review by Emiller88 — APPROVED** [review]
Looks good two things to update

File: modules/nf-core/plink2/vcf/meta.yml (hunk 3 above)
Comment (mashehu) [reply modules/nf-core/plink2/vcf/meta.yml:6]: Why should it break things? This is a valid yaml string, no?

**Comment by sateeshperi** [comment]
@nf-core-bot fix linting

**Comment by sateeshperi** [comment]
@mauro-saporita can you fix the failing tests

**Comment by mauro-saporita** [comment]
@sateeshperi ready for merging

**Pull request merged**